### PR TITLE
refactor: update WorkerExtractConfig type definition

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes

- Added required templates to `RegExManagerTemplates` and `CustomManager` for type-consistency
- `WorkerExtractConfig` now includes fields from `ExtractConfig` and `CustomExtractConfig` and some other properties necessary for extract-cache-fingerprint

## Context
- Make extract-cache-fingerprint more robust by eliminating the fields from config which do not contribute to extract logic

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
property                                  reason                                                            
 fileMatch                        can be removed since we have regexManagers included
 updateInternalDeps       code-logic is present in extract-section
 includePaths                  useful when filtering fileList
 ignorePaths                   useful when filtering fileList
 regexManagers             useful for detecting changes to custom-managers-config

File: lib/config/types.ts [hunk 1]
```diff
@@ -171,6 +174,11 @@ export interface CustomManager {
   packageNameTemplate?: string;
   versioningTemplate?: string;
   autoReplaceStringTemplate?: string;
+  depTypeTemplate?: string;
```
Comment (viceice) [inline lib/config/types.ts:177]: this type should inherited from regex templates interface instead

File: lib/config/types.ts [hunk 2]
```diff
@@ -446,10 +454,18 @@ export interface ManagerConfig extends RenovateConfig {
   language?: string | null;
 }
 
-export interface WorkerExtractConfig extends ExtractConfig {
+export interface WorkerExtractConfig
```
Comment (viceice) [inline lib/config/types.ts:457]: this type should be moved to worker where it's used

File: lib/workers/repository/extract/index.ts [hunk 3]
```diff
@@ -48,6 +48,7 @@ export async function extractAllDependencies(
   const extractResults = await Promise.all(
     extractList.map(async (managerConfig) => {
       const packageFiles = await getManagerPackageFiles(managerConfig);
+      //TODO: updateInternalDeps can be removed from WorkerExtractConfig once we move this
```
Comment (viceice) [inline lib/workers/repository/extract/index.ts:51]: not sure why 🤔

Comment (RahulGautamSingh) [reply lib/workers/repository/extract/index.ts:51]: If this option changes then deps property will change, which means that our `ExtractResult` is different from last time.

Comment (viceice) [reply lib/workers/repository/extract/index.ts:48]: i think this is no longer true with #15790, as `updateInternalDeps` is no longer used in `getManagerPackageFiles`. It's used a few lines below instead: `config.updateInternalDeps`

Comment (RahulGautamSingh) [reply lib/workers/repository/extract/index.ts:48]: It is out of the manager logic which is true.
Currently the decision is being made in extract section, but we need to move it to the lookup section where it actually is being used.
 [Need to move here](https://github.com/renovatebot/renovate/blob/5c803f2e414ac84300f037ffa6bb33dc3f558ee6/lib/workers/repository/process/fetch.ts#L29)

Comment (viceice) [reply lib/workers/repository/extract/index.ts:48]: Why it's still there, when it's no longer used? 🤔 lets remove it in a seperate PR and merge before this, so we don't need this nasty todo

File: lib/config/types.ts [hunk 4]
```diff
@@ -161,15 +160,21 @@ export type RenovateRepository =
...
   packageNameTemplate?: string;
+  datasourceTemplate?: string;
   versioningTemplate?: string;
+  depTypeTemplate?: string;
+  currentValueTemplate?: string;
+  currentDigestTemplate?: string;
+  extractVersionTemplate?: string;
+  registryUrlTemplate?: string;
+}
+export interface CustomManager extends RegexManagerTemplates {
```
Comment (viceice) [inline lib/config/types.ts:174]: we should rename this to `RegexManager`

maybe another thing to extract to separate PR. 🤔

Comment (RahulGautamSingh) [reply lib/config/types.ts:174]: Yes, will rename in another PR once we have correctly implemented narrowing as well.

Comment (viceice) [reply lib/config/types.ts:174]: Renaming should be done separately from any functional change

**Review by viceice — COMMENTED** [review]
I prefer renaming CustomManager interface in this PR

@rarkins WDYT?

**Review by viceice — APPROVED** [review]
